frontend/components/product-list/sections/FilterableProductsSection/SearchInput.tsx

```diff
@@ -1,76 +1,88 @@
 import {
-   chakra,
+   forwardRef,
    Icon,
    IconButton,
    Input,
    InputGroup,
+   InputGroupProps,
    InputLeftElement,
    InputRightElement,
 } from '@chakra-ui/react';
-import { useSearch } from '@lib/algolia';
+import { useDebounce } from '@ifixit/ui';
 import * as React from 'react';
 import { HiXCircle } from 'react-icons/hi';
 import { RiSearchLine } from 'react-icons/ri';
+import { useSearchBox } from 'react-instantsearch-hooks-web';
 
-interface SearchInputProps {
-   className?: string;
-}
+type SearchInputProps = InputGroupProps;
 
 const MAX_SEARCH_QUERY_LENGTH = 100;
+const DEBOUNCE_INTERVAL_MILLIS = 300;
 
-export const SearchInput = chakra(({ className }: SearchInputProps) => {
+export const SearchInput = forwardRef<SearchInputProps, 'div'>((props, ref) => {
+   const searchBox = useSearchBox();
    const inputRef = React.useRef<HTMLInputElement>(null);
-   const [query, search] = useSearch();
+   const [searchQuery, setSearchQuery] = React.useState(searchBox.query);
```

Comment:
Given [the docs](https://www.algolia.com/doc/api-reference/widgets/search-box/react-hooks/#hook), it's hard to tell how they expect us to use the hook.

Reply:
Do you have any specific concern here?

Reply:
Sorry, it was a lame comment. Your extracted code and the explanations are perfect. Good job solving this!